java/vector/src/main/java/org/apache/arrow/vector/util/ObjectMapperFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.util;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+/**
+ * A {@link ObjectMapper} factory to read/write JSON.
+ */
+public final class ObjectMapperFactory {
+
+  private ObjectMapperFactory() {}
+
+  /**
+   * Creates a new {@link ObjectMapper} instance.
+   */
+  public static ObjectMapper newObjectMapper() {
+    return JsonMapper.builder()
+       .addModule(new JavaTimeModule())
```

Comment:
can you add a unit test for how this will alter the behavior?

Reply:
That's the tricky part, it does not alter the the current behavior, but if you want to convince yourself this patch works, you can try to build arrow with `-Ddep.jackson.version=2.12.6` without and with the patch to observe the issue is being addressed.